### DISCUSSION TITLE
Import own python scripts errors

### DISCUSSION THREAD

**Discussion by leedchou** [nextflow-io/nextflow#2959] [Q&A] [answered]
I defined a function module in _func.py_, tried to `import func.py` in _process_ module.  The version of nextflow is _22.04_.

I am able to  import any dependencies in my env but my own python scripts. Can somebody help me with this issue?

**Comment by lungdart** [comment, marked as answer]
The python import function uses the following looks to find packages and modules to import:

1. `CWD` or the current working directory. This is the directory the python interpreter was launched from
2. the `PYTHONPATH` environment variable. This is an environment variable you can set before launching the python interpreter to add additional directories to look for packages and modules to import
3. The `sys.path` list in the program. This can be modified in the code to add additional places to look for packages and modules
4. The system, or virtual environments, globally installed packages

If you're unable to import any other scripts other than what's in your project, they likely aren't installed at all, or they aren't installed anywhere in these 4 places. The general workflow is to create a virtualenv, and use pip to install dependencies before running your scripts.

Best of luck!

**Reply by leedchou** [reply]
Thank you @lungdart for your kindly reply.

I now got it.
Adding `PYTHONPATH` in `env` scope of configuration file worked.

 
Best regards

**Reply by lungdart** [reply]
If this helped you, could you please mark the response as the accepted answer? It helps me get some fake github points. Thanks!

**Reply by leedchou** [reply]
@lungdart Yes, for sure!
